core/grid.ts

```diff
@@ -184,6 +185,23 @@ export class Grid {
     this.pattern.setAttribute('y', `${y}`);
   }
 
+  /**
+   * Given a coordinate, return the nearest coordinate aligned to the grid.
+   *
+   * @returns Object with .x and .y properties in workspace coordinates,
+   *   or null if no change.
+   */
+  alignXY(xy: Coordinate): Coordinate | null {
```

Comment:
This name seems OK but doesn't really hint at the fact that the function will return `null` if the argument was already a grid point.